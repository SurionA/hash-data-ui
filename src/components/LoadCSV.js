import React from 'react';
import Button from '@material-ui/core/Button';
import CSVReader from 'react-csv-reader';

const LoadCSV = ({ dispatch }) => {
  const loadData = (data) => {
    let i = 0;
    const columns = data[0].map((element) => {
      i++;
      return { field: `field_${i}`, headerName: element, width: 250 };
    });
    data.shift();

    let y = 0;
    const rows = data.map((row) => {
      y++;
      const datas = { id: y };
      i = 0;
      row.forEach((value) => {
        i++;
        datas[`field_${i}`] = value;
      });
      return datas;
    });

    dispatch({ columns, rows });
  };

  return (
    <label htmlFor="upload-csv">
      <CSVReader
        inputId="upload-csv"
        inputStyle={{ display: 'none' }}
        onFileLoaded={loadData}
      />
      <Button color="inherit" component="span">
        Load CSV File
      </Button>
    </label>
  );
};

export default LoadCSV;
