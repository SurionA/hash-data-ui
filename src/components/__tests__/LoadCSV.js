import React from 'react';
import renderer from 'react-test-renderer';
import CSVReader from 'react-csv-reader';
import LoadCSV from '../LoadCSV';

const mockDispatch = (object) => {
  expect(object).toEqual({
    columns: [
      { field: 'field_1', headerName: 'column1', width: 250 },
      { field: 'field_2', headerName: 'column2', width: 250 },
    ],
    rows: [{ id: 1, field_1: 'row1_field1', field_2: 'row1_field2' }],
  });
};

test('LoadCSV is being rendered', () => {
  const component = renderer.create(<LoadCSV />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Load datas', async () => {
  const component = renderer.create(<LoadCSV dispatch={mockDispatch} />);
  component.root.findByType(CSVReader).props.onFileLoaded([
    ['column1', 'column2'],
    ['row1_field1', 'row1_field2'],
  ]);
});
