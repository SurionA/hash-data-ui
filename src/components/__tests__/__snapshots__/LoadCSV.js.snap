// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadCSV is being rendered 1`] = `
<label
  htmlFor="upload-csv"
>
  <div
    className="csv-reader-input"
  >
    <input
      accept=".csv, text/csv"
      className="csv-input"
      disabled={false}
      id="upload-csv"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
  </div>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      Load CSV File
    </span>
  </span>
</label>
`;
